scalalib/src/mill/scalalib/ScalaModule.scala

```diff
@@ -23,7 +23,19 @@ trait ScalaModule extends mill.Module with TaskModule { outer =>
     override def moduleDeps = Seq(outer)
   }
   def scalaVersion: T[String]
-  def mainClass: T[Option[String]] = None
+
+  def mainClass: T[Option[String]] = T{
```

Comment:
Rather than changing `mainClass`, let's make the change to `run` and `runLocal` to check both `mainClass` and `discoverMainClasses` before running.

We should also provide a nicer error message if more than one main class is discovered. You can do so by making `run` and `runLocal` return a 

```scala
Result.Failure(
  "Multiple main classes discovered (${mains.mkString(", ")}), please override `mainClass` to specify which one you want to run"
)
```